### PR TITLE
Leian/soil migration

### PR DESCRIPTION
**Pull request by leian7**
<!--
  This is a guideline for what a PR should look like.
  Feel free to modify it to fit your specific needs.

  Please list the issue this fixes in the PR
-->
fixes issue #166 

## Changes in this PR.
<!--
  Please include a list of all things this PR will accomplish
  Use the checkbox syntax to show what is done and what is yet to be completed.
  This gives context to the diff.
-->

- [X] Modifies soil survey model
- [X] Migrates existing soil survey data from CSV into Django soil survey model objects

## Testing this PR.
<!--
  Please include a list of explicit instructions for testing this PR.
  If the instructions are 'run `make test`' say that,
  If they are more complicated be thorough.
-->

1. ``docker-compose run web bash``
2. ``cd data_scripts``
3. ``./get_sites.py``
4. ``./get_soil.py``
5. ``cd ../streamwebs_frontend``
6. ``python manage.py dbshell``
7. ``select * from streamwebs_soil_survey;``
8. Compare CSV (``/csvs/soil_survey.csv``) to table. Alternatively, go into the admin interface at localhost:8000/admin

### Expected Output.
Sample of first few rows:
<!--
  Please insert either a description of what should happen when testing
  your PR or a code-block with terminal output.
-->

```
streamwebs=# select * from streamwebs_soil_survey;
 id  |          date          | weather | landscape_pos | cover_type |   land_use   | distance |                                                                                              
                                site_char                                                                                                                               | soil_type  | school_
id | site_id 
-----+------------------------+---------+---------------+------------+--------------+----------+----------------------------------------------------------------------------------------------
------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------+--------
---+---------
 136 | 2010-08-28 18:33:00+00 |         | slope         | shrubs     | agricultural |     4.00 |                                                                                              
                                                                                                                                                                        | sand       |        
   |      36
 137 | 2010-08-28 18:45:00+00 |         | slope         | rocks      | agricultural |    11.00 | lots of ag run off                                                                           
                                                                                                                                                                        | light_clay |        
   |     189
 138 | 2010-08-11 18:22:00+00 |         | large_flat    | bare_soil  | urban        |    10.06 | Used as a parking lot for over flow events.                                                  
                                                                                                                                                                        | clay_loam  |        
   |     189
 139 | 2010-08-12 21:15:00+00 |         | stream_bank   | rocks      | wilderness   |     3.05 | dark blue water                                                                              
                                                                                                                                                                        | loamy_sand |        
   |     189
 140 | 2010-10-05 16:19:00+00 |         | stream_bank   | shrubs     | other        |     6.10 | upstream of the confluence of the Grande Ronde and Wenaha River. Land use around this site is
 a blend of agriculture, grazzing, logging, homes, wildlands and unpaved roads.                                                                                         | other      |        
   |      83

```

### CODE REVIEW

File: data_scripts/get_soil.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        elif dist == '':
+            dist = None
+        else:
+            dist = feet_to_meters(float(dist))
+        soil.distance = dist
+
+        soil.site_char = row['Distinguishing Site Characteristics']
+
+        soil_type = row['My Soil Type Is'].lower().replace(' ', '_')
+        if soil_type[len(soil_type)-1] == ')':
```
Comment (LyonesGamer) [inline data_scripts/get_soil.py]: Just so you know, you can use negative indices in Python. ``soil_type[len(soil_type)-1]`` is the same as ``soil_type[-1]``.

File: data_scripts/get_soil.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+            soil_type = soil_type[:len(soil_type)-4]
+        types = ['sand', 'loamy_sand', 'silt_loam', 'loam', 'clay_loam',
+                 'light_clay', 'heavy_clay']
+        match = False
+        for s_type in types:
+            if soil_type == s_type:         # if soil type matches valid,
+                match = True                # immediately assign
+                break                       # cancel loop
+        if match is False:                  # if doesn't match valid option,
```
Comment (LyonesGamer) [inline data_scripts/get_soil.py]: You can use an `else` clause here, actually. `for...else` causes the `else` clause to be executed once the `for` loop finishes naturally, but not if a `break` statement is used. You can get rid of the `match` variable entirely.

File: data_scripts/get_soil.py (hunk 1 above)
Comment (leian7) [inline data_scripts/get_soil.py]: @LyonesGamer Thanks for the tips!! Really good stuff to know

**Comment by Kennric** [comment]
+1

**Pull request merged**